test/functional/example_test.py

```diff
@@ -203,13 +203,13 @@ def run_test(self):
 
         # wait_until() will loop until a predicate condition is met. Use it to test properties of the
         # P2PInterface objects.
-        wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2p.block_receive_map.keys())), timeout=5, lock=mininode_lock)
+        wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2p.block_receive_map.keys())), timeout=5, lock=p2p_lock)
```

Comment:
Since you're touching this line, I'd say avoid using the `p2p` property. Especially in example_test.py
```suggestion
        wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2ps[0].block_receive_map.keys())), timeout=5, lock=p2p_lock)
```

Reply:
as above